pkg/handlers/v1/filter.go

```diff
@@ -0,0 +1,74 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/asecurityteam/awsconfig-filterd/pkg/domain"
+	"github.com/asecurityteam/awsconfig-filterd/pkg/logs"
+)
+
+// ConfigNotification is the basic structure of all incoming AWS Config SNS events
+// (https://docs.aws.amazon.com/config/latest/developerguide/example-sns-notification.html).
+type ConfigNotification struct {
+	Message            string `json:"Message"`
+	Timestamp          string `json:"Timestamp"`
+	ProcessedTimestamp string `json:"ProcessedTimestamp"`
+	Type               string `json:"Type"`
+	MessageID          string `json:"MessageId"`
+	TopicArn           string `json:"TopicArn"`
+	Subject            string `json:"Subject"`
+	SignatureVersion   string `json:"SignatureVersion"`
+	Signature          string `json:"Signature"`
+	SigningCertURL     string `json:"SigningCertURL"`
+	UnsubscribeURL     string `json:"UnsubscribeURL"`
+}
+
+// ConfigEvent represents a single AWS Config Configuration Item.
+type ConfigEvent struct {
+	ConfigurationItem domain.ConfigurationItem `json:"configurationItem"`
+}
+
+// ConfigFilterHandler applies a filter to AWS Config events.
+type ConfigFilterHandler struct {
+	Filter domain.ConfigFilter
+	LogFn  domain.LogFn
+	StatFn domain.StatFn
+}
+
+// Handle accepts Config events, applies a filter, and returns the events that match.
+func (h *ConfigFilterHandler) Handle(ctx context.Context, in ConfigNotification) (ConfigNotification, error) {
+	logger := h.LogFn(ctx)
+	stater := h.StatFn(ctx)
+
+	var event ConfigEvent
+	e := json.Unmarshal([]byte(in.Message), &event)
+	if e != nil {
+		stater.Count("event.awsconfig.filter.input_err", 1)
```

Comment:
When running as a Lambda in AWS, these metrics will be available to operators: https://docs.aws.amazon.com/lambda/latest/dg/monitoring-functions-metrics.html. When running as part of the serverfull + gateway stack then operators will have the full HTTP metrics suite for response codes and performance.

Does having a custom metric here add additional value to these?

Reply:
It does not add value, I'll remove it.

Reply:
Removed.